### ISSUE TITLE
Releases >= 2.1.3 do not launch in CentOS without disabling sandbox

### ISSUE THREAD

**Issue by apclary** [shiftkey/desktop#234]
### Describe the bug

Github Desktop does not start. No error is seen.

### Version & OS

CentOS 7

### Steps to reproduce the behavior

1. `sudo yum -y install https://github.com/shiftkey/desktop/releases/download/release-2.1.3-linux1/GitHubDesktop-linux-2.1.3-linux1.rpm`
2. Applications -> Programming -> Github Desktop
3. Nothing happens
4. Running `sudo github-desktop` also results in a silent failure.

### Expected behavior

Github Desktop UI to appear as a new window.

### Actual behavior

Nothing happened. No error message in either UI or terminal.

### Screenshots

Add screenshots to help explain your problem, if applicable.

### Logs

Attach your logs by opening the `Help` menu and selecting `Show Logs...`, if applicable.

### Additional context

Github Desktop from < 2.1.3 are able to start fine.

**Comment by shiftkey** [comment]
With the latest `2.4.0` I got a helpful clue when launching as a regular user:

```
...

Installed:
  github-desktop.x86_64 0:2.4.0_linux1-1                                        

Dependency Installed:
  libXScrnSaver.x86_64 0:1.2.2-6.1.el7  libappindicator.x86_64 0:12.10.0-13.el7 
  libdbusmenu.x86_64 0:16.04.0-4.el7    libdbusmenu-gtk2.x86_64 0:16.04.0-4.el7 
  libindicator.x86_64 0:12.10.1-6.el7  

Complete!
[shiftkey@localhost ~]$ github-desktop
Trace/breakpoint trap (core dumped)
```

Attaching `gdb` to then poke at the state of this error doesn't show anything obvious:

```
[shiftkey@localhost ~]$ gdb github-desktop
GNU gdb (GDB) Red Hat Enterprise Linux 7.6.1-115.el7
Copyright (C) 2013 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-redhat-linux-gnu".
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>...
Reading symbols from /opt/GitHub Desktop/github-desktop...(no debugging symbols found)...done.
Missing separate debuginfos, use: debuginfo-install github-desktop-2.4.0_linux1-1.x86_64
(gdb) run
Starting program: /usr/bin/github-desktop 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
[New Thread 0x7fffe28aa700 (LWP 3926)]
Detaching after fork from child process 3927.

Program received signal SIGTRAP, Trace/breakpoint trap.
0x000055555a099364 in ?? ()
Missing separate debuginfos, use: debuginfo-install alsa-lib-1.1.8-1.el7.x86_64 at-spi2-atk-2.26.2-1.el7.x86_64 at-spi2-core-2.28.0-1.el7.x86_64 atk-2.28.1-1.el7.x86_64 avahi-libs-0.6.31-19.el7.x86_64 bzip2-libs-1.0.6-13.el7.x86_64 cairo-1.15.12-4.el7.x86_64 cairo-gobject-1.15.12-4.el7.x86_64 cups-libs-1.6.3-40.el7.x86_64 dbus-libs-1.10.24-13.el7_6.x86_64 elfutils-libelf-0.176-2.el7.x86_64 elfutils-libs-0.176-2.el7.x86_64 expat-2.1.0-10.el7_3.x86_64 fontconfig-2.13.0-4.3.el7.x86_64 freetype-2.8-14.el7.x86_64 fribidi-1.0.2-1.el7_7.1.x86_64 gdk-pixbuf2-2.36.12-3.el7.x86_64 glib2-2.56.1-5.el7.x86_64 glibc-2.17-292.el7.x86_64 graphite2-1.3.10-1.el7_3.x86_64 gtk3-3.22.30-3.el7.x86_64 harfbuzz-1.7.5-2.el7.x86_64 keyutils-libs-1.5.8-3.el7.x86_64 krb5-libs-1.15.1-37.el7_7.2.x86_64 libX11-1.6.7-2.el7.x86_64 libXScrnSaver-1.2.2-6.1.el7.x86_64 libXau-1.0.8-2.1.el7.x86_64 libXcomposite-0.4.4-4.1.el7.x86_64 libXcursor-1.1.15-1.el7.x86_64 libXdamage-1.1.4-4.1.el7.x86_64 libXext-1.3.3-3.el7.x86_64 libXfixes-5.0.3-1.el7.x86_64 libXi-1.7.9-1.el7.x86_64 libXinerama-1.1.3-2.1.el7.x86_64 libXrandr-1.5.1-2.el7.x86_64 libXrender-0.9.10-1.el7.x86_64 libXtst-1.2.3-1.el7.x86_64 libattr-2.4.46-13.el7.x86_64 libblkid-2.23.2-61.el7.x86_64 libcap-2.22-10.el7.x86_64 libcom_err-1.42.9-16.el7.x86_64 libepoxy-1.5.2-1.el7.x86_64 libffi-3.0.13-18.el7.x86_64 libgcc-4.8.5-39.el7.x86_64 libgcrypt-1.5.3-14.el7.x86_64 libglvnd-1.0.1-0.8.git5baa1e5.el7.x86_64 libglvnd-egl-1.0.1-0.8.git5baa1e5.el7.x86_64 libglvnd-glx-1.0.1-0.8.git5baa1e5.el7.x86_64 libgpg-error-1.12-3.el7.x86_64 libmount-2.23.2-61.el7.x86_64 libpng-1.5.13-7.el7_2.x86_64 libselinux-2.5-14.1.el7.x86_64 libthai-0.1.14-9.el7.x86_64 libuuid-2.23.2-61.el7.x86_64 libwayland-client-1.15.0-1.el7.x86_64 libwayland-cursor-1.15.0-1.el7.x86_64 libwayland-egl-1.15.0-1.el7.x86_64 libxcb-1.13-1.el7.x86_64 libxkbcommon-0.7.1-3.el7.x86_64 lz4-1.7.5-3.el7.x86_64 nspr-4.21.0-1.el7.x86_64 nss-3.44.0-4.el7.x86_64 nss-softokn-freebl-3.44.0-5.el7.x86_64 nss-util-3.44.0-3.el7.x86_64 openssl-libs-1.0.2k-19.el7.x86_64 pango-1.42.4-4.el7_7.x86_64 pcre-8.32-17.el7.x86_64 pixman-0.34.0-1.el7.x86_64 systemd-libs-219-67.el7_7.4.x86_64 xz-libs-5.2.2-1.el7.x86_64 zlib-1.2.7-18.el7.x86_64
(gdb) 
```

Need to re-learn how to use `gdb` because it's been a while...

**Comment by shiftkey** [comment]
I thought the use of `sudo` here wasn't important, but after poking at it a bit more:

```
[shiftkey@localhost ~]$ sudo /usr/bin/github-desktop 
[sudo] password for shiftkey: 
[4399:0328/172402.197688:FATAL:atom_main_delegate.cc(211)] Running as root without --no-sandbox is not supported. See https://crbug.com/638180.
Trace/breakpoint trap
```

I then blindly followed the instructions and it launched fine:

```
[shiftkey@localhost ~]$ sudo /usr/bin/github-desktop --no-sandbox
xdg-settings: unknown desktop environment
xdg-settings: unknown desktop environment
(electron) 'getZoomFactor function' is deprecated and will be removed. Please use 'zoomFactor property' instead.
```

<img width="1025"  src="https://user-images.githubusercontent.com/359239/77834230-ec449b00-7121-11ea-9546-834d18b726be.png">


@apclary does that make any sense to you why `sudo` would change the behaviour so significantly? This feels like something we shouldn't need, but maybe that's important context to understand why it's not launching correctly without `sudo`?

**Comment by shiftkey** [comment]
This feels very similar to #222, but at the time I didn't realise the relationship between `--no-sandbox` and `sudo sysctl kernel.unprivileged_userns_clone=0`.

@apclary can you confirm that you have this setting disabled too?

```
$ sysctl kernel.unprivileged_userns_clone
kernel.unprivileged_userns_clone = 0
```

**Comment by shiftkey** [comment]
I'm going to close this out because I believe this has been fixed in [`2.4.1 RC1`](https://github.com/shiftkey/desktop/releases/tag/release-2.4.1-linux1).

**Comment by apclary** [comment]
@shiftkey Sorry for not responding. 2.4.1 did not fix it, however the --no-sandbox option did. I could launch it that way even without sudo. Thanks for that. The kernel setting just gave me "No such file or directory".

**Comment by shiftkey** [comment]
@apclary thanks for following up. I forgot to mention the additional problem after the file mode change https://github.com/shiftkey/desktop/issues/222#issuecomment-616580313 which was due to an upstream issue https://github.com/electron/electron/issues/21309 - this was related to an Electron app path containing a space affecting the Chromium sandboxing code.

That was still a problem with `2.4.1-linux1` and I updated the installation in [`2.4.1 RC2`](https://github.com/shiftkey/desktop/releases/tag/release-2.4.1-linux2) to move the installed app under `/usr/lib/github-desktop`. That should avoid the need to `--no-sandbox` in the future.

EDIT: sigh, I missed that this was related to the RPM install, which remain unresolved. The good news is that this will be fixed by #273 and I'll queue up an update during the week so we can verify you're good.
